### PR TITLE
Fix broken models and add CI

### PR DESCRIPTION
**Pull request by rok-cesnovar**
- fixes broken models (not compiling): 
   - knitr/bandits/bernoulli-ab.stan
   - knitr/car-iar-poisson/pois_icar.stan
   - knitr/neural-nets/nn.stan

I think the fixes are correct (based on similar models in the subfolders).

Adds Continous integration that will try to compile all modified or added models.

### CODE REVIEW

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    
+    - uses: actions/checkout@v2
+      with:
+        path: 'example-models-new'
+        fetch-depth: 0
+    
+    - name: Write modified models
+      run: |
+        cd example-models-new
+        echo "example-models-new/basic_distributions/binormal.stan" > ../test-models.txt
```
Comment (rok-cesnovar) [inline .github/workflows/compile.yml:33]: This is just so we always compile at least one model (if changes in PR dont touch models).

Comment (bbbales2) [reply .github/workflows/compile.yml:33]: Are tests here only running when the test models change?

That doesn't make sense to me. Shouldn't the tests run if the code changes? We do not expect the models to change much I thought.

Comment (rok-cesnovar) [reply .github/workflows/compile.yml:33]: These tests are for PRs/pushes to example models repo. So we dont add broken models from now on.

Comment (rok-cesnovar) [reply .github/workflows/compile.yml:33]: They are cloned and compiled/run in other repos (like stanc3 tests will use them here: https://github.com/stan-dev/stanc3/blob/master/Jenkinsfile#L149 once the broken tests are fixed).

Comment (bbbales2) [reply .github/workflows/compile.yml:33]: Oooo, got it

**Review by bbbales2 — APPROVED** [review]
Good!

**Pull request merged**